repos/tap-resolver/src/builders/buildAssets.js

```diff
@@ -11,97 +36,214 @@ const tapConstants = require('../tap/tapConstants')
 
-  // Check that the path exists, and is a directory
-  return checkTapAssetPath && isDirectory(checkTapAssetPath, true)
-    ? { full: checkTapAssetPath, relative: tapAssets }
-    : { full: defAssetsPath, relative: assetsPath }
+  // Build the fonts path relative to the tap
+  const checkTapFontsPath = isStr(tapFonts) && path.join(TAP_PATH, tapFonts)
+
+  // Set the defaults for the assets and fonts paths
+  const foundPaths = {
+    defAssetsPath,
+    defFontsPath,
+    fontsFull: defFontsPath,
+    assetsFull: defAssetsPath,
+    fontsRelative: fontsPath,
+    assetsRelative: assetsPath,
+  }
+
+  // If the assets path exists, update the foundPaths with the taps asset paths
+  checkTapAssetPath &&
+    isDirectory(checkTapAssetPath, true) &&
+    Object.assign(foundPaths, {
+      assetsRelative: tapAssets,
+      assetsFull: checkTapAssetPath,
+    })
+
+  // If the fonts path exists, update the foundPaths with the taps fonts paths
+  checkTapFontsPath &&
+    isDirectory(checkTapFontsPath, true) &&
+    Object.assign(foundPaths, {
+      fontsRelative: tapFonts,
+      fontsFull: checkTapFontsPath,
+    })
+
+  return foundPaths
+}
+
+/**
+ * Checks if a file has the .js extension, and adds it if needed
+ * @param {string} file - File name to check for the .js extension
+ *
+ * @returns {string} - file with the .js extension added
+ */
+const ensureJsExt = file => (path.extname(file) === '.js' ? file : `${file}.js`)
+
+/**
+ * Write the assets and font imports to a valid javascript file
+ * @param {string} assetsFull - Path to the taps asset folder
+ * @param {string} fontsFull - Path to the taps fonts folder
+ * @param {string} assetIndex - Name of the assets index file
+ * @param {string} fontsIndex - Name of the fonts index file
+ * @param {string} builtAssets - auto-generated imports of asset files as string
+ * @param {string} builtFonts - Assets with a font extension
+ *
+ * @returns {void}
+ */
+const writeAssetFiles = args => {
+  const {
+    assetIndex,
+    fontsIndex,
+    assetsPath,
+    fontsPath,
+    builtAssets,
+    builtFonts,
+  } = args
+
+  // Write the assets index file to the assets/index location
+  fs.writeFileSync(
+    path.join(assetsPath, ensureJsExt(assetIndex)),
+    autoGeneratedText.concat(
+      `\nexport const assets = {\n${builtAssets.join(',\n')}\n}\n`
+    )
+  )
+
+  // Write the fonts file to the assets/fonts location
+  fs.writeFileSync(
+    path.join(fontsPath, ensureJsExt(fontsIndex)),
+    autoGeneratedText.concat(
+      `\nexport const fonts = {\n${builtFonts.join(',\n')}\n}\n`
+    )
+  )
+}
+
+/**
+ * Validates that a file path does not contain spaces
+ * React-Native does not allow assets or fonts in the path name
+ * @param {string} assetName - Name of the asset to check
+ *
+ * @returns {string} - Passed in assetName if it does not contain spaces
+ */
+const validateNoSpace = assetName => {
+  if (!assetName.includes(' ')) return assetName
```

Comment:
are other types of whitespace a problem with RN as well? Like tab characters? If so you might want to use a regex test with the general whitespace pattern `/\s+/g`

Reply:
This checks the name of a file on disk.
Any other `whitespace` characters are not allowed within file names, so I don't think it's needed.
Although it may be possible to hack an invalid character into a file name, I think the user would have bigger problems then if it can be loaded into the keg.